### PR TITLE
#167313397 Email verification

### PR DESCRIPTION
**Pull request by habinezadalvan**
#### What does this PR do?

- Verifies if the email used by the user when signing up or admin creating other admins exists.

#### Description of Task to be completed?

- Created the email template that has to be sent to the user registering to Authors' heaven.

- Created a url (route) that verifies the user.

- Created a middleware that prevent a user to access other routes when he/she didn't confirm the email.

#### How should this be manually tested?

1. Clone the repo

2. Run <npm install>

3. Create .env file, and  include the credentials found in .env.sample file

4. Run <npm run migration>

5. Use postman to test sign up 

6. Go the email used when signing up and verify the received email.

#### Any background context you want to provide?

- none

#### What are the relevant pivotal tracker stories?

 [167313397](https://www.pivotaltracker.com/story/show/167313397)

#### Screenshots (if appropriate)

Screen Shot 2019-08-08 at 12 10 46

Screen Shot 2019-08-08 at 12 11 39

Screen Shot 2019-08-08 at 12 10 33

Screen Shot 2019-08-08 at 12 12 05

#### Questions:

@Mnickii @Anguandia @minega25 @nshutijonathan @nkalyesubula @salviosage

### CODE REVIEW

File: src/middlewares/admin.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-      });
-    }
-    next();
-  };
-  export default admin;
\ No newline at end of file
+  if (!(req.auth.role === 'admin')) {
+    return res.status(403).send({
+      status: 403,
+      message: 'Access denied.Only admin'
```
Comment (Mnickii) [inline src/middlewares/admin.js]: please include a more readable message

File: src/routes/api/user/user.route.js [hunk 2]
```diff
@@ -3,14 +3,22 @@ import UserController from '../../../controllers/user.controller';
 import validateToken from '../../../middlewares/auth';
 import validateUser from '../../../middlewares/validators/signup.validation';
 import admin from '../../../middlewares/admin';
+import verifyEmail from '../../../controllers/verify-controller';
+// import confirmEmaiAuth from '../../../middlewares/emailVarification.middleware';
```
Comment (Mnickii) [inline src/routes/api/user/user.route.js]: remove comment

**Review by nkalyesubula — APPROVED** [review]
Good Leon. This looks fine to me

**Review by Mnickii — CHANGES_REQUESTED** [review]

File: src/middlewares/admin.js (hunk 1 above)
Comment (habinezadalvan) [reply src/middlewares/admin.js]: I worked on the feedback; and now It's done.

**Review by Mnickii — APPROVED** [review]
LGTM

**Review by minega25 — APPROVED** [review]

**Comment by minega25** [comment]
Great job.

**Pull request merged**